java/impl/gcp/src/main/java/co/worklytics/psoxy/CloudFunctionRequest.java

```diff
@@ -92,4 +94,30 @@ public byte[] getBody() {
     public String prettyPrint() {
         return request.toString();
     }
+
+
+    public List<String> getWarnings() {
+        List<String> warnings = new LinkedList<>();
+
+        //Compression-related
+        // clients should request compressed content to reduce running costs of proxy instances;
+        // warn if they aren't
+        // https://cloud.google.com/appengine/docs/legacy/standard/go111/how-requests-are-handled#:~:text=For%20responses%20that%20are%20returned,HTML%2C%20CSS%2C%20or%20JavaScript.
+        warnIfHeaderAbsentOrMissingGzip(HttpHeaders.ACCEPT_ENCODING).ifPresent(warnings::add);
+        warnIfHeaderAbsentOrMissingGzip(HttpHeaders.USER_AGENT).ifPresent(warnings::add);
```

Comment:
Yeah and it seems that depending on the value of the user agent it may be compressed or not